### PR TITLE
Add login, lobby, new game, and in-game screen specific requirements

### PR DESCRIPTION
**Pull request by devinmcginty**
About halfway through writing this I realized that I had no idea what the difference between a functional requirement and a use case is. Does anybody have any input?

### CODE REVIEW

File: requirements/specific_requirements.rst [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+New Game Screen
+```````````````
+- The application allows a user to choose whether they wish to play black, white, or no preference (defaults no preference).
+- The application allows the user to invite another user to play as the opponent.
+- The application provides a button that creates the new game instance and takes the user to an In-Game Screen of the new game.
+
+In-Game Screen
+```````````
+- If the user is viewing a completed game the application displays a model dialogue with the final game state.
```
Comment (gnidan) [inline requirements/specific_requirements.rst]: modal*

**Comment by gnidan** [comment]
@devinmcginty I'd say that a functional requirement should be something like:

- "The application allows the user to do _blank_"
- "The system behaves by _blank_ when _blank_"

Or for the move capabilities stuff, maybe just make a section like "Specific Game Piece Valid Moves" and explain that each piece has specific requirements for how it behaves. Or something like that.

I guess basically I see the difference as... 

- Functional requirements are how the system behaves, organized by groupings of features.
- Use cases are how the user achieves specific actions/behaviors and how the system changes while these happen. So organized by these high-level flows.

Make sense?

**Comment by gnidan** [comment]
That being said, this looks on the right track for sure. I'd say just think about it like you're writing a check-list for us later.

**Comment by devinmcginty** [comment]
I believe I added everything required for the user interface. However, I believe we should also add in the functional requirements for the back end (how the database will be set up and how requests will be sent). I'm not sure how we are implementing this. Should I write something simple for now, or does somebody want to add a new section?

**Comment by gnidan** [comment]
Gonna say this LGTM for now!

**Pull request merged**